libs/island-ui/core/src/lib/Text/Text.stories.mdx

```diff
@@ -0,0 +1,157 @@
+import { Meta, Story, Canvas, Props } from '@storybook/addon-docs/blocks'
+import { Text } from './Text'
+import { Box } from '../Box'
+import { select } from '@storybook/addon-knobs'
+import { Colors } from '@island.is/island-ui/theme'
+import { VariantTypes } from './Text.treat'
+import { GridColumn, GridContainer, GridRow } from '../Grid'
+import { LinkContext } from '../LinkContext'
+
+<Meta title="Core/Text" component={Text} />
+
+# Text
+
+## Our general text component
+
+<br />
+
+### Control the appearance by choosing a `variant`
+
+<Canvas>
+  <Story name="Variants">
+    <Text>
+      This is the default <code>(p)</code> variant
+    </Text>
+    <Text variant="h1">
+      This is the <code>h1</code> variant
+    </Text>
+    <Text variant="h2">
+      This is the <code>h2</code> variant
+    </Text>
+    <Text variant="h3">
+      This is the <code>h3</code> variant
+    </Text>
+    <Text variant="h4">
+      This is the <code>h4</code> variant
+    </Text>
+    <Text variant="h5">
+      This is the <code>h5</code> variant
+    </Text>
+    <Text variant="intro">
+      This is the <code>intro</code> variant
+    </Text>
+    <Text variant="small">
+      This is the <code>small</code> variant
+    </Text>
+    <Text variant="eyebrow">
+      This is the <code>eyebrow</code> variant
+    </Text>
+  </Story>
+</Canvas>
+
+### Customize it with lineHeight/fontWidth
+
+<Canvas>
+  <Story name="Customize">
+    <Text variant="h3" lineHeight="lg" fontWeight="light">
+      This is the 'h3' variant with a different line-height and font-weight
+    </Text>
+  </Story>
+</Canvas>
+
+### Control the rendered element by specifying `as`
+
+<Canvas>
+  <Story name="Element">
+    <Text variant="h3" as="h1">
+      <code>h3</code> variant and renders as <code>{'<h1/>'}</code> in the DOM
+    </Text>
+  </Story>
+</Canvas>
+
+### Custom links inside text
+
+Depending on which React router we're using we often need to take control of how links are rendered.
+Wrap your app in `<LinkContext.Provider />` and provide a custom renderer function.
+
+#### Example:
+
+<Canvas>
+  <Story name="Custom link renderer">
+    <LinkContext.Provider
+      value={{
+        linkRenderer: (href, children) => (
+          <a
+            style={{
+              color: 'deeppink',
+              fontFamily: 'Comic Sans MS',
+              fontWeight: 500,
+              textShadow: '1px 1px 0 rebeccapurple',
+            }}
+            href={href}
+            rel="noopener noreferrer"
+            target="_blank"
+          >
+            {children + '✨'}
```

Comment:
We should have an eslint rule for this. I've seen `'{children}✨'` and ``{children + '✨'}``. Would probably better to only have one way to write it imho

Reply:
You should never do this in real life JEREMY